spec/integration/knife/client_create_spec.rb

```diff
@@ -0,0 +1,69 @@
+#
+# Copyright:: Copyright 2013-2016, Chef Software Inc.
+# License:: Apache License, Version 2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+require "support/shared/integration/integration_helper"
+require "support/shared/context/config"
+require "openssl"
+
+describe "knife client create", :workstation do
+  include IntegrationSupport
+  include KnifeSupport
+
+  include_context "default config options"
+
+  let(:out) { "Created client[bah]\n" }
+
+  when_the_chef_server "is empty" do
+    it "creates a new client" do
+      knife("client create -k bah").should_succeed stderr: out
+    end
+
+    it "creates a new validator client" do
+      knife("client create -k --validator bah").should_succeed stderr: out
+      knife("client show bah").should_succeed <<EOM
+admin:     false
+chef_type: client
+name:      bah
+validator: true
+EOM
+    end
+
+    it "refuses to add an existing client" do
+      pending "Knife client create must not blindly overwrite an existing client"
+      knife("client create -k bah").should_succeed stderr: out
+      expect { knife("client create -k bah") }.to raise_error(Net::HTTPServerException)
+    end
+
+    it "saves the private key to a file" do
+      Dir.mktmpdir do |tgt|
+        knife("client create -f #{tgt}/bah.pem bah").should_succeed stderr: out
+        expect(File).to exist("#{tgt}/bah.pem")
+      end
+    end
+
+    it "reads the public key from a file" do
```

Comment:
not easily - zero doesn't store the default key since it doesn't really care about it